File: src/go/partition/s/small.go
package small

import (
	parallelNP "golang/common"
	partition "golang/partition/lib"
	"os"
	"time"
)

func Small(arr [][]int, verbose bool, outS string, outP string, outST string, outPT string) {
	// Sequential
	start_seq := time.Now()
	resultS := Seq(arr)
	end_seq := time.Since(start_seq)

	// Parallel
	start_par := time.Now()
	resultP := Par(arr)
	end_par := time.Since(start_par)

	// Write the result to the file
	if verbose {
		fileS, err := os.Create(outS)
		if err != nil {
			parallelNP.IOError("large.go", "Error creating the file", err)
		}
		defer fileS.Close()
		partition.WritePartitions(fileS, resultS)
		timeS, err := os.Create(outST)
		if err != nil {
			parallelNP.IOError("large.go", "Error creating the file", err)
		}
		defer timeS.Close()
		parallelNP.WriteTime(timeS, end_seq)

		fileP, err := os.Create(outP)
		if err != nil {
			parallelNP.IOError("large.go", "Error creating the file", err)
		}
		defer fileP.Close()
		partition.WritePartitions(fileP, resultP)
		timeP, err := os.Create(outPT)
		if err != nil {
			parallelNP.IOError("large.go", "Error creating the file", err)
		}
		defer timeP.Close()
		parallelNP.WriteTime(timeP, end_par)
	}
}

File: src/julia/partition/s/small.jl
module Small

include("./seq.jl")
using .Seq
include("./par.jl")
using .Par
include("../partitions2json.jl")
using .Partitions2Json
include("../../common/write.jl")
using .Write

export small
function small(arr::Vector{Vector{Int64}}, verbose::Bool, outS::String, outP::String, outST::String, outPT::String)
    # Sequential
    start_seq = Base.time_ns()
    resultS = Seq.small_seq(arr)
    end_seq = Base.time_ns()
    
    # Parallel
    start_par = Base.time_ns()
    resultP = Par.small_par(arr)
    end_par = Base.time_ns()


    if verbose
        file = open(outS, "w")
        partitions2json(file, resultS)
        close(file)
        timeS = open(outST, "w")
        writeTime(timeS, (end_seq - start_seq) / 1e9) # Convert to seconds
        close(timeS)

        file = open(outP, "w")
        partitions2json(file, resultP)
        close(file)
        timeP = open(outPT, "w")
        writeTime(timeP, (end_par - start_par) / 1e9) # Convert to seconds
        close(timeP)
    end

end # small

end # module